### DISCUSSION TITLE
Graphics.drawPath / Graphics.curveTo doesn't work as expected

### DISCUSSION THREAD

**Discussion by ylazy** [airsdk/Adobe-Runtime-Support#808] [Q&A]
Hello!
I'm trying to draw a 50x50px rectangle (**cornerRadius = 7**) using drawPath:
```
var commands:Vector.<int> = new Vector.<int>();
    commands.push(1, 2, 3, 2, 3, 2, 3, 2, 3);

var data:Vector.<Number> = new Vector.<Number>();
    data.push(
    7, 0,
    43, 0,
    50, 0, 50, 7,
    50, 43,
    50, 50, 43, 50,
    7, 50,
    0, 50, 0, 43,
    0, 7,
    0, 0, 7, 0);
	
var s:Shape = new Shape();
    s.graphics.beginFill(0xFF0000);
    s.graphics.drawPath(commands, data);
    s.graphics.endFill();
    s.x = 100;
    s.y = 100;
    this.addChild(s);
```
I got the result (600% zoomed in)
![image](https://user-images.githubusercontent.com/6010183/115580770-9a088600-a2f1-11eb-8175-c53cc3503d2c.png)

As you can see, the 4 corners have different corner radius:
```
Top-Left: 7x5px
Top-Right: 7x5px
Bottom-Left: 5x5px
Bottom-Right: 5x5px
```

Is that a bug?
Could anyone please tell me how to fix it?
Thank you so much!

**Comment by ajwfrost** [comment]
Can you provide the commands list please? and what renderMode setting are you using?
thanks

**Reply by ylazy** [reply]
I've updated my post for the commands list.
I'm using renderMode = auto.
Thank you!

**Comment by ajwfrost** [comment]
Thanks - I think the challenge here would be that changing anything in the rendering engine would impact a lot of existing apps! so if you want a curve of a certain look, you may need to adjust the commands a little or perhaps use a line style with a curved corner?

**Reply by ylazy** [reply]
Hi Andrew,
Could you please tell me how to adjust the commands? I'm building UI Components, so I can't use lineStyle.
My opinion is, if it's a bug, it should be fixed even if it would impact a lot of existing apps.
In other words, it should be fixed asap because it affects a lot of apps. And the change will make a lot of apps look better.

**Comment by ylazy** [comment]
This issue also happens with **drawRoundedRect, drawCircle, curveTo**.
I've tried to draw corners using drawCircle and got a similar result.
There's no workaround.
